src/Sarif.Viewer.VisualStudio.UnitTests/SdkUiUtilitiesTests.cs

```diff
@@ -0,0 +1,197 @@
+            };
+
+            var actual = SdkUiUtilities.GetInlinesForErrorMessage(message, data: 1, clickHandler: Hyperlink_Click);
+
+            actual[0].Should().BeOfType(expected[0].GetType());
+            (actual[0] as Run).Text.Should().Be((expected[0] as Run).Text);
+        }
+
+        [Fact]
+        public void SarifSnapshot_GetMessageEmbeddedLinkInlines_OneLink()
+        {
+            string message = @"The quick [brown fox](1) jumps over the lazy dog.";
+
+            var link = new Hyperlink();
+            link.Tag = new Tuple<object, int>(1, 1);
+            link.Inlines.Add(new Run("brown fox"));
+
+            var expected = new List<Inline>
+            {
+                new Run("The quick "),
+                link,
+                new Run(" jumps over the lazy dog.")
+            };
+
+            var actual = SdkUiUtilities.GetInlinesForErrorMessage(message, data: 1, clickHandler: Hyperlink_Click);
+
+            actual.Count.Should().Be(expected.Count);
+
+            actual[0].Should().BeOfType(expected[0].GetType());
+            (actual[0] as Run).Text.Should().Be((expected[0] as Run).Text);
+
+            actual[1].Should().BeOfType(expected[1].GetType());
+            (actual[1] as Hyperlink).Inlines.Count.Should().Be((expected[1] as Hyperlink).Inlines.Count);
+            ((actual[1] as Hyperlink).Inlines.FirstInline as Run).Text.Should().Be(((expected[1] as Hyperlink).Inlines.FirstInline as Run).Text);
+            Tuple<object, int> tagActual = (actual[1] as Hyperlink).Tag as Tuple<object, int>;
+            Tuple<object, int> tagExpected = (actual[1] as Hyperlink).Tag as Tuple<object, int>;
+            tagActual.Item1.Should().Be(tagExpected.Item1);
+            tagActual.Item2.Should().Be(tagExpected.Item2);
+
+            actual[2].Should().BeOfType(expected[2].GetType());
+            (actual[2] as Run).Text.Should().Be((expected[2] as Run).Text);
+        }
+
+        [Fact]
+        public void SarifSnapshot_GetMessageEmbeddedLinkInlines_TwoLinksAndEndsWithLink()
+        {
+            string message = @"The quick [brown fox](1) jumps over the [lazy dog](2)";
+
+            var link1 = new Hyperlink();
+            link1.Tag = new Tuple<object, int>(1, 1);
+            link1.Inlines.Add(new Run("brown fox"));
+
+            var link2 = new Hyperlink();
+            link2.Tag = new Tuple<object, int>(1, 2);
+            link2.Inlines.Add(new Run("lazy dog"));
+
+            var expected = new List<Inline>
+            {
+                new Run("The quick "),
+                link1,
+                new Run(" jumps over the "),
+                link2
+            };
+
+            var actual = SdkUiUtilities.GetInlinesForErrorMessage(message, data: 1, clickHandler: Hyperlink_Click);
+
+            actual.Count.Should().Be(expected.Count);
+
+            actual[0].Should().BeOfType(expected[0].GetType());
+            (actual[0] as Run).Text.Should().Be((expected[0] as Run).Text);
```

Comment:
Consider: For these two lines, introduce a helper
```
private static void VerifyTextRun(Inline expected, Inline actual)
```
... so this would be
```
VerifyTextRun(actual[0], expected[0]);
```